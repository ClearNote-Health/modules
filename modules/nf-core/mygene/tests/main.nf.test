nextflow_process {

    name "Test Process MYGENE"
    script "../main.nf"
    process "MYGENE"

    tag "modules"
    tag "modules_nfcore"
    tag "mygene"

    test("mygene - txt") {
        when {
            process {
                """
                input[0] = [
                    [id : 'test'],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.gene_meta.tsv")
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tsv).match("tsv") },
                { assert snapshot(process.out.gmt).match("gmt") }
            )
        }
    }
}
