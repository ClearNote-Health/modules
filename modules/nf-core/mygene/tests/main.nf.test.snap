{
    "mygene - filter by go evidence - versions": {
        "content": [
            [
                "versions.yml:md5,09d72645c3ae7e886af6e8bd2876c72b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:47:18.062632"
    },
    "mygene - default options - versions": {
        "content": [
            [
                "versions.yml:md5,09d72645c3ae7e886af6e8bd2876c72b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:46:42.027592"
    },
    "mygene - default with tsv file - versions": {
        "content": [
            [
                "versions.yml:md5,09d72645c3ae7e886af6e8bd2876c72b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:46:54.221092"
    },
    "mygene - default options - gmt": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gmt:md5,4af358fb273e9702e82c4cf012dd168d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:46:42.013225"
    },
    "mygene - filter by go category - versions": {
        "content": [
            [
                "versions.yml:md5,09d72645c3ae7e886af6e8bd2876c72b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:47:05.668961"
    },
    "mygene - filter by go evidence - gmt": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gmt:md5,e76fdf742b36064d639c3fed3c610ff5"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:47:18.051678"
    },
    "mygene - default with tsv file - tsv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.tsv:md5,018e23173b224cbf328751006593900e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:46:54.214475"
    },
    "mygene - default with tsv file - gmt": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gmt:md5,4af358fb273e9702e82c4cf012dd168d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:46:54.207568"
    },
    "mygene - filter by go category - gmt": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gmt:md5,6d4061d6d9eb93693245c8d5583e14ac"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T16:47:05.655456"
    }
}